core/random.rbs

```diff
@@ -81,8 +134,9 @@ class Random < Object
   # (`-`) and add (`+`)methods, or rand will raise an ArgumentError.
   #
   def rand: () -> Float
-          | (Integer | ::Range[Integer] max) -> Integer
+          | (Integer | ::Range[Integer] | Numeric max) -> Integer
```

Comment:
Seems like having `Numeric` here will return `Integer` for any numbers including `Float`.
Moving this line after next line would give better type.